versioned/mongodb/src/main/java/com/dremio/nessie/versioned/store/mongodb/MongoDbStore.java

```diff
@@ -0,0 +1,247 @@
+    collections.put(ValueType.L3, mongoDatabase.getCollection(MongoDbConstants.L3_COLLECTION, L3.class));
+    collections.put(ValueType.COMMIT_METADATA,
+        mongoDatabase.getCollection(MongoDbConstants.COMMIT_METADATA_COLLECTION, InternalCommitMetadata.class));
+    collections.put(ValueType.KEY_FRAGMENT,
+        mongoDatabase.getCollection(MongoDbConstants.KEY_FRAGMENT_COLLECTION, Fragment.class));
+    collections.put(ValueType.REF, mongoDatabase.getCollection(MongoDbConstants.REF_COLLECTION, InternalRef.class));
+    collections.put(ValueType.VALUE,
+        mongoDatabase.getCollection(MongoDbConstants.VALUE_COLLECTION, InternalValue.class));
+  }
+
+  /**
+   * Closes the connection this manager creates to a database. If the connection is already closed this method has
+   * no effect.
+   */
+  @Override
+  public void close() {
+    if (null != mongoClient) {
+      mongoClient.close();
+    }
+  }
+
+  @Override
+  public void load(LoadStep loadstep) throws ReferenceNotFoundException {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public <V> boolean putIfAbsent(ValueType type, V value) {
+    final MongoCollection collection = collections.get(type);
+    if (null == collection) {
+      throw new UnsupportedOperationException(String.format("Unsupported Entity type: %s", type.name()));
+    }
+
+    // TODO: Potentially rework to use updateOne with upsert and $set.
+    try (MongoCursor cursor = collection.find(Filters.eq(Store.KEY_NAME, ((HasId) value).getId())).iterator()) {
+      if (!cursor.hasNext()) {
+        collection.insertOne(value);
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  @Override
+  public <V> void put(ValueType type, V value, Optional<ConditionExpression> conditionUnAliased) {
+    Preconditions.checkArgument(type.getObjectClass().isAssignableFrom(value.getClass()),
+        "ValueType %s doesn't extend expected type %s.", value.getClass().getName(), type.getObjectClass().getName());
+
+    final MongoCollection collection = collections.get(type);
+    if (null == collection) {
+      throw new UnsupportedOperationException(String.format("Unsupported Entity type: %s", type.name()));
+    }
+
+    // TODO: Correct this so it overwrites values if already present.
+    // TODO: Throw an IllegalArgumentException (or something) if the ConditionExpression can't be satisfied.
+    collection.insertOne(value);
+  }
+
+  @Override
+  public boolean delete(ValueType type, Id id, Optional<ConditionExpression> condition) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void save(List<SaveOp<?>> ops) {
+    // TODO: Should these operations be async?
+    final ListMultimap<MongoCollection<?>, SaveOp<?>> mm = Multimaps.index(ops, l -> collections.get(l.getType()));
+    final ListMultimap<MongoCollection<?>, InsertOneModel<?>> writes =
+        Multimaps.transformValues(mm, s -> new InsertOneModel<>(s.getValue()));
+
+    if (enableTransactions) {
```

Comment:
We should not be using any transactions.

Reply:
Transactions removed.